### ISSUE TITLE
Criticall error when using KeepObjects function

### ISSUE THREAD

**Issue by dimethylglymaxim** [SchmollerLab/Cell_ACDC#218]
### Description:

Whenever I try to use Keep Objects, a critical error emerges. The function is working nonetheless. 
[2022-12-02_09-28-20_gui_stdout.log](https://github.com/SchmollerLab/Cell_ACDC/files/10141447/2022-12-02_09-28-20_gui_stdout.log)

### Way to reproduce:

1. Use the Keep Objects function with multiple objects to keep.

### Terminal output:

```Shell
Traceback (most recent call last):
  File "C:\Users\myermolaie\Miniconda3\envs\acdc\lib\site-packages\cellacdc\__init__.py", line 100, in inner_function
    result = func(self)
  File "C:\Users\myermolaie\Miniconda3\envs\acdc\lib\site-packages\cellacdc\gui.py", line 15033, in applyKeepObjects
    removeAnnot = result[1]
TypeError: 'bool' object is not subscriptable
```

**Comment by ElpadoCan** [comment]
Hi @dimethylglymaxim, thank you for reporting this, I will fix it soon.

You are right, the function will still work because that error is raised after the objects are removed. 

What actually fails is the propagation of the edits to future frames, but I already saw why. I will keep you updated.

Cheers,
Francesco

**Comment by ElpadoCan** [comment]
Hi @dimethylglymaxim I released `v1.2.4rc55` which fixes this issue. Let me know if it works, thanks! 

Cheers,
Francesco

**Comment by dimethylglymaxim** [comment]
Hey @ElpadoCan 

It propagates as it should, but now it deletes all masks in the next frames (still, the frame on which the function was activated is properly liberated of unwanted masks). And critical error message do not emerge anymore either.

Cheers,
Max

**Comment by ElpadoCan** [comment]
Hi Max, if you propagate to future frames then it is correct that it deletes the non kept IDs, right?

**Comment by dimethylglymaxim** [comment]
No, it deletes all IDs, even those that should be kept.

**Comment by ElpadoCan** [comment]
You are absolutely right, I will release a fix soon, thanks for reporting this!

**Comment by ElpadoCan** [comment]
Hi Max,

I just release `v1.2.4rc56` which should have a fix for this issue.

Please update with `pip install --upgrade cellacdc` (after activating the acdc environment) and let me know if it works, thanks! 

Have a good Christmas break and a happy new year! 

Cheers, 
Francesco

**Comment by dimethylglymaxim** [comment]
Hi! 
Thank you for your work! 

The function operates as intended, but there is one minor bug left:
1. Movie, 90 frames. First frame - 3 cells: IDs 1, 2, and 3.
2. On the frame, for example, 10 - fourth cell appears (as well as other cells, ID n): ID 4.
3. When I use this function looking at the first frame (1, 2, 3) and specify in "objects to keep" IDs 1, 2, 3, 4, then in the later frames, only IDs 1, 2, and 3 will be present. 

In summary:
If the ID specified in the function parameter does not exist on a frame on which this function was triggered, it will be deleted.

Anyway, again, thank you for all the work you've done. I wish you a wonderful Christmas and a happy new year too!

**Comment by ElpadoCan** [comment]
Wow, that's a sneaky one I didn't think about it, thank you very much for spotting it, very appreciated! I will fix it soon! 

Cheers,
Francesco

**Comment by ElpadoCan** [comment]
Hi Max, 

I found that you can "cheat" and go to select the cells that you want to keep in the frame where they exist, then without deactivating the function, you navigate to the first frame and apply the deletion. This way it works. Of course, it is not ideal so I will fix it, but at least for now, you have a workaround.

Cheers,
Francesco
